service/src/main.rs

```diff
@@ -149,6 +144,18 @@ fn main() {
 		println!("Worker Config: {:?}", config);
 		let skip_ra = smatches.is_present("skip-ra");
 		let dev = smatches.is_present("dev");
+		let set_interval = smatches.is_present("interval");
```

Comment:
"interval" is not a flag, but an input value. So it's somewhat confusing to use it as a flag.

I'd recommend to do it like it's done for the w-server:

```rust
smatches
    .value_of("w-server")
    .map(ToString::to_string)
    .unwrap_or_else(|| format!("ws://127.0.0.1:{}", config.worker_rpc_port)),
```
This makes it clear, that in case if it's not set, a default value is used. What I like about your approach, however, is that in case of a wrong input, it does not continue. 👍 

Maybe a mix of both approached is possible? Use default value in case it's not set, but panic in case it set wrongly?